shell/platform/common/cpp/client_wrapper/event_channel_unittests.cc

```diff
@@ -0,0 +1,179 @@
+  const StandardMethodCodec& codec = StandardMethodCodec::GetInstance();
+  EventChannel channel(&messenger, channel_name, &codec);
+
+  bool on_listen_called = false;
+  bool on_cancel_called = false;
+  flutter::StreamHandler<flutter::EncodableValue> handler(
+      [&on_listen_called](
+          const flutter::EncodableValue* arguments,
+          std::unique_ptr<flutter::EventSink<flutter::EncodableValue>>&&
+              events) { on_listen_called = true; },
+      [&on_cancel_called](const flutter::EncodableValue* arguments) {
+        on_cancel_called = true;
+      });
+  channel.SetStreamHandler(&handler);
+  EXPECT_EQ(messenger.last_message_handler_channel(), channel_name);
+  EXPECT_NE(messenger.last_message_handler(), nullptr);
+
+  // Send dummy listen message.
+  MethodCall<flutter::EncodableValue> call_listen("listen", nullptr);
+  auto message = codec.EncodeMethodCall(call_listen);
+  messenger.last_message_handler()(
+      message->data(), message->size(),
+      [](const uint8_t* reply, const size_t reply_size) {});
+  EXPECT_EQ(on_listen_called, true);
+
+  // Send dummy cancel message.
+  MethodCall<flutter::EncodableValue> call_cancel("cancel", nullptr);
+  message = codec.EncodeMethodCall(call_cancel);
+  messenger.last_message_handler()(
+      message->data(), message->size(),
+      [](const uint8_t* reply, const size_t reply_size) {});
+
+  // Check results.
+  EXPECT_EQ(on_cancel_called, true);
+}
+
+// Test that consecutive call of OnListen.
+TEST(EventChannelTest, ConsecutiveListen) {
+  TestBinaryMessenger messenger;
+  const std::string channel_name("some_channel");
+  const StandardMethodCodec& codec = StandardMethodCodec::GetInstance();
+  EventChannel channel(&messenger, channel_name, &codec);
+
+  bool on_listen_called = false;
+  bool on_cancel_called = false;
+  flutter::StreamHandler<flutter::EncodableValue> handler(
+      [&on_listen_called](
+          const flutter::EncodableValue* arguments,
+          std::unique_ptr<flutter::EventSink<flutter::EncodableValue>>&&
+              events) { on_listen_called = true; },
+      [&on_cancel_called](const flutter::EncodableValue* arguments) {
+        on_cancel_called = true;
+      });
+  channel.SetStreamHandler(&handler);
+  EXPECT_EQ(messenger.last_message_handler_channel(), channel_name);
+  EXPECT_NE(messenger.last_message_handler(), nullptr);
+
+  // Send dummy listen message.
+  MethodCall<flutter::EncodableValue> call("listen", nullptr);
+  auto message = codec.EncodeMethodCall(call);
+  messenger.last_message_handler()(
+      message->data(), message->size(),
+      [](const uint8_t* reply, const size_t reply_size) {});
+  EXPECT_EQ(on_listen_called, true);
+
+  // Send dummy listen message.
+  message = codec.EncodeMethodCall(call);
+  messenger.last_message_handler()(
+      message->data(), message->size(),
+      [](const uint8_t* reply, const size_t reply_size) {});
+  EXPECT_EQ(on_listen_called, true);
```

Comment:
You didn't set it to false, after the previous assert, so this line isn't actually testing anything.

Reply:
Thank you. I added `on_listen_called` reset.